services/jobservices/src/condor-launcher/condor.go

```diff
@@ -0,0 +1,425 @@
+//
+// condor-launcher launches jobs on an HTCondor cluster.
+//
+// This service connects to an AMQP broker's "jobs" exchange and waits for
+// messages sent with the "jobs.launches" key. It then turns the request
+// into an iplant.cmd, config, job, and irods_config file in /tmp/<user>/<UUID>
+// and calls out to condor_submit to submit the job to the cluster.
+//
+// Since it launches jobs by executing the condor_submit command it shouldn't
+// run inside a Docker container. Our Condor cluster is moderately large and
+// requires a lot of ports to be opened up, which doesn't play nicely with
+// Docker.
+//
+// Required configuration keys are:
+//   amqp.uri
+//   irods.user
+//   irods.pass
+//   irods.host
+//   irods.port
+//   irods.base
+//   irods.resc
+//   irods.zone
+//   condor.condor_config
+//   condor.path_env_var
+//   condor.log_path
+//   condor.request_disk
+//   porklock.image
+//   porklock.tag
+//
+package main
+
+import (
+	"bytes"
+	"configurate"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"logcabin"
+	"messaging"
+	"model"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"text/template"
+
+	"github.com/streadway/amqp"
+)
+
+var (
+	logger  = logcabin.New("condor-launcher", "condor-launcher")
+	cfgPath = flag.String("config", "", "Path to the config file. Required.")
+	version = flag.Bool("version", false, "Print the version information")
+	gitref  string
+	appver  string
+	builtby string
+)
+
+func init() {
+	flag.Parse()
+}
+
+// GenerateCondorSubmit returns a string (or error) containing the contents
+// of what should go into an HTCondor submission file.
+func GenerateCondorSubmit(submission *model.Job) (string, error) {
```

Comment:
Is it necessary to export this function? Same goes for `GenerateJobConfig`, `GenerateIRODSConfig`, `CreateSubmissionDirectory`, and `CreateSubmissionFiles`.